Library/Homebrew/sbom.rb

```diff
@@ -0,0 +1,354 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cxxstdlib"
+require "json"
+require "development_tools"
+require "extend/cachable"
+require "utils/curl"
+
+# Rather than calling `new` directly, use one of the class methods like {SBOM.create}.
+class SBOM
+  extend Cachable
+
+  FILENAME = "sbom.spdx.json"
+  SCHEMA = "https://raw.githubusercontent.com/spdx/spdx-spec/v2.3/schemas/spdx-schema.json"
+
+  attr_accessor :homebrew_version, :spdxfile, :built_as_bottle, :installed_as_dependency, :installed_on_request,
+                :changed_files, :poured_from_bottle, :loaded_from_api, :time, :stdlib, :aliases, :arch, :source,
+                :built_on, :license, :name
+  attr_writer :compiler, :runtime_dependencies, :source_modified_time
+
+  # Instantiates a {SBOM} for a new installation of a formula.
+  sig { params(formula: Formula, compiler: T.nilable(String), stdlib: T.nilable(String)).returns(T.attached_class) }
+  def self.create(formula, compiler = nil, stdlib = nil)
```

Comment:
```suggestion
  def self.create(formula, compiler: nil, stdlib: nil)
```

makes callers read more nicely because not obvious why these would be what they are.